tests/cases/fourslash/autoImportCompletionExportEqualsWithDefault1.ts

```diff
@@ -0,0 +1,91 @@
+/// <reference path="fourslash.ts" />
+
+// @strict: true
+// @module: commonjs
+
+// @filename: node.ts
+//// import Container from "./container.js";
+//// import Document from "./document.js";
+////
+//// declare namespace Node {
+////   class Node extends Node_ {}
+////
+////   export { Node as default };
+//// }
+////
+//// declare abstract class Node_ {
+////   parent: Container | Document | undefined;
+//// }
+////
+//// declare class Node extends Node_ {}
+////
+//// export = Node;
+
+// @filename: document.ts
+//// import Container from "./container.js";
+////
+//// declare namespace Document {
+////   export { Document_ as default };
+//// }
+////
+//// declare class Document_ extends Container {}
+////
+//// declare class Document extends Document_ {}
+////
+//// export = Document;
+
+// @filename: container.ts
+//// import Node from "./node.js";
+////
+//// declare namespace Container {
+////   export { Container_ as default };
+//// }
+////
+//// declare abstract class Container_ extends Node {
+////   p/*1*/
+//// }
+////
+//// declare class Container extends Container_ {}
+////
+//// export = Container;
+
+const preferences = {
+  includeCompletionsWithInsertText: true,
+  includeCompletionsWithClassMemberSnippets: true,
+};
+
+verify.completions({
+  marker: "1",
+  includes: [
+    {
+      name: "parent",
+      insertText: "parent: Container_ | Document_ | undefined;",
+      filterText: "parent",
+      hasAction: true,
+      source: "ClassMemberSnippet/",
+    },
+  ],
+  preferences,
+  isNewIdentifierLocation: true,
+});
+
+verify.applyCodeActionFromCompletion("1", {
+  name: "parent",
+  source: "ClassMemberSnippet/",
+  description: `Includes imports of types referenced by 'parent'`,
+  newFileContent: `import Document_ from "./document.js";
```

Comment:
What happens if we're in `--module nodenext` and the importing file is `.mjs` while document.js is still CJS?

Reply:
This test case is circular and, from what I remember, that was somewhat important to trigger the bug. Since it's circular I kinda have to stay within bounds of a single module system.

I tried to repro it by introducing a new `.mts` file and it works fine even today:
```ts
// @filename: node.ts
import Container from "./container.cjs";
import Document from "./document.cjs";

declare namespace Node {
  class Node extends Node_ {}

  export { Node as default };
}

declare abstract class Node_ {
  parent: Container | Document | undefined;
}

declare class Node extends Node_ {}

export = Node;

// @filename: document.cts
import Container from "./container.cjs";

declare namespace Document {
  export { Document_ as default };
}

declare class Document_ extends Container.default {}

declare class Document extends Document_ {}

export = Document;

// container.cts
import Node from "./node.cjs";

declare namespace Container {
  export { Container_ as default };
}

declare abstract class Container_ extends Node.default {}

declare class Container extends Container_ {}

export = Container;

// @filename: index.mts
import Container from './container.cjs'

declare abstract class MyCls extends Container.default {
    p/*1*/
}
```

I'm not sure if this is a test case that you are looking for. The new content of the `index.mts` file here becomes:
```ts
import Container from './container.cjs'

declare abstract class MyCls extends Container.default {
    parent: Container | import("./document.cjs") | undefined;
}
```